app/src/components/search/filters/selectFilterModal.tsx

```diff
@@ -0,0 +1,239 @@
+                  name: "filter_keywords",
+                  placeholder: `Search ${filter.name.toLowerCase()}${
+                    filter.name === "Publishers" ? "" : "s"
+                  }`,
+                  onChange: (e) => setSearchText(e.target.value),
+                }}
+                onSubmit={() => {}}
+              />
+              <Box
+                sx={{
+                  marginTop: "s",
+                  border: "1px solid",
+                  borderColor: "ui.border.hover",
+                  padding: "s",
+                  height: "394px",
+                }}
+              >
+                <RadioGroup
+                  isFullWidth
+                  // Helper invalid text, even when false, creates a 8px margin.
+                  sx={{ marginBottom: "-8px" }}
+                  id={`${filter.name}-options`}
+                  labelText={`${filter.name} filter options`}
+                  showLabel={false}
+                  name={filter.name}
+                  defaultValue={selected?.name ?? ""}
+                  showHelperInvalidText={false}
+                  onChange={(newValue) => {
+                    const newSelectedOption = filter.options.find(
+                      (option) => option.name === newValue
+                    );
+                    setSelected(newSelectedOption || null);
+                  }}
+                >
+                  {radioFilterOptions(filter.name, currentOptions)}
+                </RadioGroup>
+              </Box>
+              <Flex>
+                <Pagination
+                  id="filter-pagination-id"
+                  currentPage={currentPage}
+                  initialPage={1}
+                  pageCount={pageCount}
+                  onPageChange={(page) => {
+                    setCurrentPage(page);
+                  }}
+                  sx={{
+                    paddingTop: "m",
+                    justifyContent: "center",
+                    svg: {
+                      _visited: {
+                        _disabled: { fill: "ui.disabled.primary !important" },
+                      },
+                    },
+
+                    a: {
+                      _disabled: {
+                        color: "ui.disabled.primary",
+                        pointerEvents: "none",
+                        svg: { fill: "ui.disabled.primary !important" },
+                      },
+                    },
+                  }}
+                />
+              </Flex>
+            </ModalBody>
+
+            <ButtonGroup padding="m" marginX="auto">
+              <Button
+                buttonType="secondary"
+                onClick={handleClose}
+                id="close-button"
+              >
+                Close
+              </Button>
+              <Button
+                id="confirm-button"
+                isDisabled={!selected}
+                onClick={() => {
+                  setSelected(
```

Comment:
This is not working as expected. The `onChange` in the RadioGroup already updates the selected value so this "Confirm" button doesn't really confirm the selected choice, it really just closes the modal. You can replicate this by selecting any unselected option and then clicking on "Close". I would expect to not have any value selected but when I open the same select, the value is actually selected.

Separately, but related, when I click "Close", the focus goes to the "Displaying 1-48 of .... results" text but it shouldn't. I did not make a selection that would update the result list, so the focus should go back to the "View all [name]" button.